back/engines/commercial/analytics/app/services/analytics/query_schema.json

```diff
@@ -0,0 +1,117 @@
+{
+  "type" : "object",
+  "$schema": "http://citizenlab.co/tenant_settings_spec.json",
+  "properties" : {
+    "fields": {
+      "anyOf" : [
+        {
+          "type" : "string",
+          "pattern" : "^[a-z_]*\.[a-z_]*$"
+        }, 
+        {
+          "type": "array",
+          "items" : {
+            "type" : "string",
+            "pattern" : "^[a-z_]*\.[a-z_]*$"
+          },
+          "minLength" : 1
+        }
+      ]
+    },
+    "fact" : {
+      "type" : "string",
+      "enum" : ["post", "participation"]
+    },
+    "dimensions" : {
+      "type" : "object",
+      "minProperties" : 1,
+      "patternProperties" : {
+        "^[a-z_]*$" : {
+          "type" : "object",
+          "minProperties" : 1,
+          "patternProperties" : {
+            "^[a-z_]*$" : {
+              "type" : ["string", "array", "object"],
```

Comment:
Is it valid json-schema to specify multiple types? didn't know :)

Reply:
a bit surprised that we can use required properties if the property can be a string or an array tho.

Reply:
yes you can specify multiple types and also specify all posible keys that are going to validate compatible values.
so for example if you have an array, the key `properties` is ignored, and the key `minLength` is applied.

Reply:
OK, good to know :)